### PR TITLE
Endpoint for APIv4: GET {siteUrl}/files/{file_id}/public

### PR DESCRIPTION
**Pull request by saturninoabril**
Summary

APIv4 (independent): GET {siteUrl}/files/{file_id}/public

Ticket Link

Jira ticket: [PLT-5247](https://mattermost.atlassian.net/browse/PLT-5247)

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (required for all new APIs) [none but related docs for public file link generation](https://github.com/mattermost/mattermost-api-reference/pull/125)
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

File: app/file.go [hunk 1]
```diff
@@ -360,7 +360,7 @@ func MigrateFilenamesToFileInfos(post *model.Post) []*model.FileInfo {
 
 func GeneratePublicLink(siteURL string, info *model.FileInfo) string {
 	hash := GeneratePublicLinkHash(info.Id, *utils.Cfg.FileSettings.PublicLinkSalt)
-	return fmt.Sprintf("%s%s/public/files/%v/get?h=%s", siteURL, model.API_URL_SUFFIX_V3, info.Id, hash)
+	return fmt.Sprintf("%s/files/%v/public?h=%s", siteURL, info.Id, hash)
```
Comment (jwilander) [inline app/file.go]: I think you'll actually need to make another function that's for v4 only, that way we can still use this one with v3 (hence the unit test failure)

Comment (saturninoabril) [inline app/file.go]: actually, I did such backwards compatibility in the earlier [PR# 5607](https://github.com/mattermost/platform/pull/5607/files) but was just repeated the function here for test purpose only. I intend to rebase once PR# 5607 is merged.
I've seen the build failure of this PR but not sure if it has something to do with the changes I've made. I'll take a look later. thanks

Comment (saturninoabril) [inline app/file.go]: @jwilander please ignore this for a while. i'll get back to this later once i returned to my dev pc. thanks

**Comment by saturninoabril** [comment]
@jwilander please ignore this for a while. i'll get back to this later once i returned to my dev pc. thanks

**Comment by grundleborg** [comment]
Hi @saturninoabril , would you be able to rebase this onto current master? Thanks.

**Comment by saturninoabril** [comment]
@grundleborg done rebase. thanks

**Pull request merged**